dht.go

```diff
@@ -228,29 +228,17 @@ func (dht *IpfsDHT) getValueSingle(ctx context.Context, p peer.ID, key string) (
 // getLocal attempts to retrieve the value from the datastore
 func (dht *IpfsDHT) getLocal(key string) (*recpb.Record, error) {
 	log.Debugf("getLocal %s", key)
-
-	v, err := dht.datastore.Get(mkDsKey(key))
+	rec, err := dht.getRecordFromDatastore(mkDsKey(key))
 	if err != nil {
 		return nil, err
 	}
-	log.Debugf("found %s in local datastore")
 
-	byt, ok := v.([]byte)
-	if !ok {
-		return nil, errors.New("value stored in datastore not []byte")
```

Comment:
This `not []byte` error and the `unmarshal` error below are now going to be logged but the function will return `nil` in those cases. Whereas for the case where the key does not match `ErrNotFound` will be returned. Should all three cases return `ErrNotFound`?

Reply:
Yes. Sorry, it should return nil, nil.